### PR TITLE
WIP Add conversion functions to get JmsMessages from jms.Messages

### PR DESCRIPTION
**Pull request by codingismy11to7**
This makes it easy to get the Alpakka `JmsMessage`s with the body, all headers, and all JMS properties intact.

This is a first pass, I only have so much time to work on this so wanted to get some discussion before going any further with anything else that will be required.

## Purpose

`JmsProducer` requires Alpakka `JmsMessage` model objects to produce messages in a flow (or plain body objects with no headers in a sink). While implementing a standard request/reply model, I found there was no automatic way to convert a `jms.Message` to a `JmsMessage` while keeping headers & properties intact (need fields such as `correlationId` and `replyTo` from the original message).

I added the missing standard JMS headers in `JmsHeader`, and added conversions from `jms.[Map|Text|Bytes|Object]Message` to their corresponding `JmsMessage` instances.

## References

https://discuss.lightbend.com/t/alpakka-activemq-request-response-with-pooled-temporary-queues/4012/9

### CODE REVIEW

**Comment by ennru** [comment]
This looks very good. At first glance, I didn't see anything to object on.

You might want to add something to the docs which initiated this PR as I wrote down in https://github.com/akka/alpakka/issues/1782 to show this PR in context.

**Comment by codingismy11to7** [comment]
Removed the `@author` comment IDEA added and added documentation

File: jms/src/test/java/docs/javadsl/JmsConnectorsTest.java [hunk 1]
```diff
@@ -868,6 +869,73 @@ public void passThroughEmptyMessageEnvelopes() throws Exception {
...
+          // #request-reply
+          JmsConsumerControl respondStreamControl =
+              JmsConsumer.create(
+                      JmsConsumerSettings.create(system, connectionFactory).withQueue("test"))
+                  .map(JmsMessage::create)
+                  .collectType(JmsTextMessage.class)
+                  .map(
+                      textMessage -> {
+                        JmsTextMessage m = JmsTextMessage.create(reverse.apply(textMessage.body()));
+                        for (JmsHeader h : JavaConverters.asJavaCollection(textMessage.headers()))
```
Comment (ennru) [inline jms/src/test/java/docs/javadsl/JmsConnectorsTest.java]: Oh, I wasn't aware that there is no Java API to access the message headers, properties and destination. We should add that in another change (see https://github.com/akka/alpakka/issues/1800).

**Comment by ennru** [comment]
Looks good, but has compilation errors on both Scala 2.11 and 2.13:

https://travis-ci.com/akka/alpakka/jobs/211931446#L977

https://travis-ci.com/akka/alpakka/jobs/211931447#L2127

**Comment by ennru** [comment]
I pushed fixes for the Scala versions compatibility and added a Java API to make the use from Java nicer.

**Pull request merged**